src/main/java/nextstep/subway/line/domain/PathGraph.java

```diff
@@ -0,0 +1,80 @@
+package nextstep.subway.line.domain;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import nextstep.subway.line.dto.PathResponse;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.domain.StationRepository;
+import nextstep.subway.station.dto.StationResponse;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PathGraph {
+
+    private final StationRepository stationRepository;
+
+    public PathGraph(StationRepository stationRepository) {
+        this.stationRepository = stationRepository;
+    }
+
+    public PathResponse findPath(Long source, Long target, List<Line> lines){
```

Comment:
```suggestion
    public PathResponse findPath(Station source, Station target, List<Section> sections){
```
`Line`은 필요없지 않을까요? 😀